### PR TITLE
Fix that try sample and show adv option buttons could be clicked without Git URL in different import flows

### PR DESCRIPTION
**Pull request by jerolimov**
**Fixes**: 
https://issues.redhat.com/browse/ODC-5955
https://issues.redhat.com/browse/ODC-6016

**Analysis / Root cause**: 
With 4.8 we automatically focus the Git URL (the first input field) when the user imports from Git, Dockerfile or Devfile.

When the user clicks then on "Try sample" or "Show advanced Git options" the url field losts his focus. The `onBlur` method then marks the input field as `touched` (in formik) and show an error if the url field is empty.

This "Required!" error was shown faster then the mentioned buttons handles their event. And because the "Required!" was added as a new label below the input field all buttons and input fields jumps a little but down and the user click on these buttons is ignored.

A second click works fine. Also if the user first lost the focus of the input field, and then select it again.

**Solution Description**: 
Add a help text to the "Git Repo URL" field. This help text was replaced with the error message when the form field is empty (or invalid).

Also fixed a small flicking because `git.isUrlValidating` was set to false before the validation status was set.

The help text for the different cases are:

1. Repository URL to build and deploy your code from source.
2. Repository URL to build and deploy your code from a Dockerfile.
3. Repository URL to build and deploy your code from a Devfile.

Solution was discussed with UX in slack [#tmp-odc-add](https://app.slack.com/client/T027F3GAJ/CHC2R6AGG/thread/CHC2R6AGG-1623436371.040500)

**Screen shots / Gifs for design review**: 
Before:
Peek 2021-06-17 15-56

After:
Peek 2021-06-17 15-57

**Unit test coverage report**: 
Just added a new help text. Unit tests are not changed.

**Test setup:**
* Open developer perspective
* Add > Import from Devfile > Click "Try sample"
* Add > Import from Git > Click "Show advanced options"

**Browser conformance**: 
- [x] Chrome
- [x] Firefox
- [x] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/packages/dev-console/src/components/import/git/GitSection.tsx [hunk 1]
```diff
@@ -260,6 +259,18 @@ const GitSection: React.FC<GitSectionProps> = ({
         'devconsole~URL is valid but cannot be reached. If this is a private repository, enter a source Secret in advanced Git options',
       );
     }
+    if (validated === ValidatedOptions.error && buildStrategy === BuildStrategyType.Devfile) {
+      return t('devconsole~No Devfile present, unable to continue');
+    }
+    if (buildStrategy === BuildStrategyType.Source) {
+      return t('devconsole~Repository URL to built and deploy your code from source.');
```
Comment (debsmita1) [inline frontend/packages/dev-console/src/components/import/git/GitSection.tsx:266]:
```suggestion
      return t('devconsole~Repository URL to build and deploy your code from source.');
```

same for the remaining 2 help texts

File: frontend/packages/dev-console/src/components/import/git/GitSection.tsx [hunk 2]
```diff
@@ -260,6 +259,18 @@ const GitSection: React.FC<GitSectionProps> = ({
         'devconsole~URL is valid but cannot be reached. If this is a private repository, enter a source Secret in advanced Git options',
       );
     }
+    if (validated === ValidatedOptions.error && buildStrategy === BuildStrategyType.Devfile) {
+      return t('devconsole~No Devfile present, unable to continue');
```
Comment (debsmita1) [inline frontend/packages/dev-console/src/components/import/git/GitSection.tsx:263]:
```suggestion
      return t('devconsole~No Devfile present, unable to continue.');
```

**Review by debsmita1 — CHANGES_REQUESTED** [review]
Tried out the changes and it looks good. Left a few comments related to the help texts. Will wait for UX to approve the changes

File: frontend/packages/dev-console/src/components/import/git/GitSection.tsx (hunk 1 above)
Comment (jerolimov) [reply frontend/packages/dev-console/src/components/import/git/GitSection.tsx]: Thanks. Updated.

File: frontend/packages/dev-console/src/components/import/git/GitSection.tsx (hunk 2 above)
Comment (jerolimov) [reply frontend/packages/dev-console/src/components/import/git/GitSection.tsx]: Updated.

**Comment by jerolimov** [comment]
> Will wait for UX to approve the changes

It is approved by @beaumorley and @parvathyvr in this [slack thread](https://app.slack.com/client/T027F3GAJ/CHC2R6AGG/thread/CHC2R6AGG-1623436371.040500), but maybe one of you both could add a comment to this PR as well. Thanks :)

**Pull request merged**